commit 5407486090ccb3aed9a8bd92706f8ddb6e56a6b8
Author: Elliot Saba <staticfloat@gmail.com>
Date:   Sun Mar 23 14:05:19 2025 +0000

    gcc/config/darwin.h: Backport `-gstabs` removal
    
    This backports some specific changes in [0] to disable passing `-gstabs`
    to the assembler on macOS, as that is not supported in the newer
    assemblers anymore.
    
    [0] https://github.com/gcc-mirror/gcc/commit/7e0db0cdf01e9c885a29cb37415f5bc00d90c029#diff-113bc1be804c926d135d2d563c03164a3885aa944944d737bb249c01b6087c78L589

diff --git a/gcc/config/darwin.h b/gcc/config/darwin.h
index f017353778b..894ec76b3f9 100644
--- a/gcc/config/darwin.h
+++ b/gcc/config/darwin.h
@@ -466,16 +466,11 @@ extern GTY(()) int darwin_ms_struct;
 /* Default ASM_DEBUG_SPEC.  Darwin's as cannot currently produce dwarf
    debugging data.  */
 
-#define ASM_DEBUG_SPEC  "%{g*:%{%:debug-level-gt(0):%{!gdwarf*:--gstabs}}}"
+#define ASM_DEBUG_SPEC  ""
 #define ASM_FINAL_SPEC \
   "%{gsplit-dwarf:%ngsplit-dwarf is not supported on this platform} %<gsplit-dwarf"
 
-/* We still allow output of STABS if the assembler supports it.  */
-#ifdef HAVE_AS_STABS_DIRECTIVE
-#define DBX_DEBUGGING_INFO 1
-#define PREFERRED_DEBUGGING_TYPE DBX_DEBUG
-#endif
-
+#define PREFERRED_DEBUGGING_TYPE DWARF2_DEBUG
 #define DWARF2_DEBUGGING_INFO 1
 
 #define DEBUG_FRAME_SECTION	  "__DWARF,__debug_frame,regular,debug"
